src/components/xeniumQC/XeniumLabwareQC.tsx

```diff
@@ -16,71 +20,145 @@ type XeniumLabwareQCProps = {
-                      onWorkNumberChange={(workNumber) => {
-                        setFieldValue(`labware.${index}.workNumber`, workNumber);
+              </div>
+              <div className="flex flex-row items-center justify-start"></div>
+              <div className={'grid grid-cols-2 gap-4'}>
+                <WorkNumberSelect
+                  label={'SGP Number'}
+                  name={`labware.${index}.workNumber`}
+                  dataTestId={`${labware.barcode}-workNumber`}
+                  onWorkNumberChange={async (workNumber) => {
+                    await setFieldValue(`labware.${index}.workNumber`, workNumber);
+                  }}
+                  workNumber={values.labware[index]?.workNumber}
+                />
+
+                <CustomReactSelect
+                  label={'Comments'}
+                  dataTestId={`${labware.barcode}-comments`}
+                  name={`labware.${index}.comments`}
+                  emptyOption={true}
+                  options={selectOptionValues(comments, 'text', 'id')}
+                  handleChange={async (val) => {
+                    await setFieldValue(
+                      `labware.${index}.comments`,
+                      (val as OptionType[]).map((option) => option.value)
+                    );
+                  }}
+                  isMulti={true}
+                  value={values.labware[index]?.comments?.map(
+                    (commentId) => comments.find((comment) => comment.id === Number(commentId))?.text
+                  )}
+                />
+              </div>
+              <div>
+                <header className="text-lg font-bold py-8">Region of interest</header>
+              </div>
+              {!values.labware[index]?.sampleComments && (
+                <Warning data-testid={'warning'} message={'No regions of interest recorded for this labware.'} />
+              )}
+              {values.labware[index]?.sampleComments?.length > 0 && (
+                <div className="grid grid-cols-6 gap-4">
+                  <div className="col-span-2">
+                    <CustomReactSelect
+                      label={'Apply to all'}
+                      dataTestId={`labware.${index}.roi-comments`}
+                      name={`labware.${index}.roiComments`}
+                      emptyOption={true}
+                      options={selectOptionValues(comments, 'text', 'id')}
+                      handleChange={async (val) => {
+                        const comments = (val as OptionType[]).map((option) => option.value);
+                        await setValues((prev) => {
+                          return {
+                            ...prev,
+                            labware: prev.labware.map((labware, i) => {
+                              if (i === index) {
+                                return {
+                                  ...labware,
+                                  roiComments: comments,
```

Comment:
I presume that `sampleComments` is written to match the form in the graphql request, but is `roiComments` something separate? What is it?

Reply:
setValues() method is linked to the UI form, roiComments refers to the global comment component(apply to all), I am adjusting the form properties to the graphql request before sending it.